### PR TITLE
Feature/order

### PR DESCRIPTION
**Pull request by rauljch9**

### CODE REVIEW

File: src/model/orders/Order.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+public class Order {
+    Integer id;
+    Client client;
+    Collection<OrderDetail> orderDetailList;
+    String note;
+
+    /**
+     * @return The price of the list of products
+     */
+    protected Double getProductListPrice(){
```
Comment (reynoldmorel) [inline src/model/orders/Order.java:22]: Why is this protected? Why is this here? I think this should be outside of the class

File: src/model/orders/Order.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public Collection<OrderDetail> getOrderDetailList() {
+        return orderDetailList;
+    }
+
+    public void setOrderDetailList(Collection<OrderDetail> orderDetailList) {
+        this.orderDetailList = orderDetailList;
+    }
+
+    public void addOrderDetail(OrderDetail orderDetail) {
```
Comment (reynoldmorel) [inline src/model/orders/Order.java:61]: You don't need to do this since you are exposing `orderDetails`

File: src/model/orders/Order.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    /**
+     * @return this method calculates the estimated preparation time of the whole order
+     */
+    public Double getEstimatedPreparationTime(){
+        return null;
+    }
+
+    public Integer getId() {
```
Comment (reynoldmorel) [inline src/model/orders/Order.java:37]: Id should be long

File: src/model/orders/Order.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import java.util.Collection;
+
+/**
+ * The Order Class has this Client and the list of OrderDetail (list of product and quantity)
+ */
+public class Order {
+    Integer id;
```
Comment (reynoldmorel) [inline src/model/orders/Order.java:14]: These variables should be private

File: src/model/orders/Order.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    protected Double getProductListPrice(){
+        Double totalPrice = 0.0;
+        for (OrderDetail orderDetail: orderDetailList) {
+            totalPrice+=orderDetail.getProduct().getPrice();
+        }
+        return totalPrice;
+    }
+
+    /**
+     * @return this method calculates the estimated preparation time of the whole order
```
Comment (reynoldmorel) [inline src/model/orders/Order.java:31]: What is this for? It's not calculating anything, don't add code that you are not going to use right now.

**Review by reynoldmorel — APPROVED** [review]

File: src/model/orders/Order.java (hunk 2 above)
Comment (rauljch9) [reply src/model/orders/Order.java:61]: Got it, I will fix it

File: src/model/orders/Order.java (hunk 3 above)
Comment (rauljch9) [reply src/model/orders/Order.java:37]: Woulg you tell me why it shoul be **Long** so I know if a have to change the **id** field in all the other model classes?

Because I am using **Integer** in all aproved model classes, I am even using int in some classes.

File: src/model/orders/Order.java (hunk 4 above)
Comment (rauljch9) [reply src/model/orders/Order.java:14]: Same here!

Shoul I change the access modifier in the field **id** in all my approved model classes?

File: src/model/orders/Order.java (hunk 1 above)
Comment (rauljch9) [reply src/model/orders/Order.java:22]: I had it private because I wanted the subclasses to call this methos instead of outside classes, but know that it's not abstract, I have to fix it

File: src/model/orders/Order.java (hunk 5 above)
Comment (rauljch9) [reply src/model/orders/Order.java:31]: I'll fix it
